packages/fluentui/react-bindings/src/FocusZone/FocusZone.tsx

```diff
@@ -220,12 +235,12 @@ export class FocusZone extends React.Component<FocusZoneProps> implements IFocus
 
   componentWillUnmount() {
     delete _allInstances[this._id];
+    outZones.unregister(this.windowElement!, this);
 
     if (!this._isInnerZone) {
-      _outerZones.delete(this);
-
-      if (this.windowElement && _outerZones.size === 0) {
+      if (this.windowElement && outZones.getOutZone(this.windowElement)?.size === 0) {
         this.windowElement.removeEventListener('keydown', _onKeyDownCapture, true);
+        outZones.deleteOutZone(this.windowElement!);
```

Comment:
I'm wondering why the outerZone unregister doesn't do this.

Reply:
Recommend moving this to last unregistration and removing the api. (Because it does this work on first registration, but on last unregistration it doesn't. This is asymmetrical.)